oauthproxy.go

```diff
@@ -1149,6 +1107,22 @@ func extractAllowedGroups(req *http.Request) map[string]struct{} {
 	return groups
 }
 
+// encodedState builds the OAuth state param out of our nonce and
+// original application redirect
+func encodeState(csrf cookies.CSRF, redirect string) string {
+	return fmt.Sprintf("%v:%v", csrf.HashOAuthState(), redirect)
+}
+
+// decodeState splits the reflected OAuth state response back into
+// the nonce and original application redirect
+func decodeState(req *http.Request) (string, string, error) {
```

Comment:
The encode (which I assume is meant to be opposite of this) takes a `cookies.CSRF` in, I semi expected this to return one in that case, is that not a useful thing to do here? Maybe this would be clearer if the encode just accepted the nonce as a string and HashOAuthState was called by the caller?

Reply:
Did we want to resolve this comment?